### DISCUSSION TITLE
Setting cookies during SSR?

### DISCUSSION THREAD

**Discussion by lephyrius** [leptos-rs/leptos#1499] [Q&A] [answered]
I'm using SSR (rendering in default mode) I have a server function that sets a cookie in Actix. 
But it seems like it isn't set. Is it possible?
What is the recommended solution?

**Comment by gbj** [comment, marked as answer]
If you providing an example of some code, people could try to help debug. Otherwise, we can only guess. My guess is that you should use `create_blocking_resource` for the resource where you're calling the server function that writes the cookie, because otherwise the headers of the HTTP response will already have been sent by the time that Suspense resolves and it will be too late.

**Reply by lephyrius** [reply]
I got my answer. I think 🤔 I just go with a fetch and set the cookie in that way. I need the data but I don’t need to block the entire page render in order to do that. I could just do it lazily when user press the button.

**Reply by NicTanghe** [reply]
Hello could you post a snippet

**Reply by benwis** [reply]
> Hello could you post a snippet 

Please create a new post and don't  necroba year old one

**Reply by benwis** [reply]
> Hello could you post a snippet 

Please create a new post and don't  necroba year old one
